kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/OpenShiftStartupStrategy.java

```diff
@@ -0,0 +1,198 @@
+                                    ZonedDateTime.now().format(DateTimeFormatter.ISO_INSTANT));
+                            client.deploymentConfigs().createOrReplace(dc);
+
+                            logger.info("Update DeploymentConfig: {}", md.getName());
+                        } else {
+                            logger.info("Event - Ignored");
+                        }
+                    }
+
+                    @Override
+                    public void onClose(KubernetesClientException cause) {
+                        logger.info("Watch Closed.");
+                        if (cause != null) {
+                            logger.info(cause.getMessage());
+                        }
+                    }
+                })) {
+                    logger.info("Created watch.");
+
+                    Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+                        synchronized (this) {
+                            isWatchRunning = false;
+                            notifyAll();
+                            logger.info("ShutdownHook sent notifyAll.");
+                        }
+                    }));
+
+                    synchronized (this) {
+                        while (isWatchRunning && !Thread.currentThread().isInterrupted()) {
+                            logger.info("WatchRunner starts.");
+                            try {
+                                wait();
+                            } catch (InterruptedException e) {
+                                Thread.currentThread().interrupt();
+                                logger.error("Thread interrupted", e);
+                            }
+                            logger.info("WatchRunner notified.");
+                        }
+                        logger.info("WatchRunner exits.");
+                    }
+                }
+            } catch (KubernetesClientException e) {
+                logger.error("Failed", e);
+            }
+        }
+    }
+
+    @Override
+    public void startup(KieServerImpl kieServer,
+                        ContainerManager containerManager,
+                        KieServerState currentState,
+                        AtomicBoolean kieServerActive) {
+
+        String kieServerId = currentState.getConfiguration()
+                                         .getConfigItem(KieServerConstants.KIE_SERVER_ID).getValue();
+
+        try (OpenShiftClient client = (OpenShiftClient) createAdaptiveClient()) {
+
+            Set<KieContainerResource> containers = prepareContainers(currentState.getContainers());
+            containerManager.installContainersSync(kieServer, containers, currentState, new KieServerSetup());
+
+            new Thread(new WatchRunner(kieServerId)).start();
+
+            triggerRollout(client, kieServerId, true);
+        }
+    }
+
+    private static boolean triggerRollout(OpenShiftClient client, String kieServerId, boolean completed) {
```

Comment:
@rhtevan Can you split this method?
Right now it contains 2 different functionalities, differentiated by completed flag.

Reply:
Committed change as per suggested.